qp_deblur/deblur.py

```diff
@@ -0,0 +1,112 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import join
+
+from future.utils import viewitems
+from functools import partial
+from collections import OrderedDict
+
+from qiita_client import ArtifactInfo
+from qiita_client.util import system_call
+
+
+def generate_deblur_workflow_commands(preprocessed_fastq, out_dir, parameters):
+    """Generates the deblur commands
+
+    Parameters
+    ----------
+    preprocessed_fastq : str
+        The preprocessed_fastq filepaths
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    list of str
+        The deblur commands
+
+    Raises
+    ------
+    ValueError
+        If there is more than 1 file passed as preprocessed_fastq
+    """
+    if len(preprocessed_fastq) != 1:
+        raise ValueError("deblur doesn't accept more than one FASTQ: "
+                         "%s" % ', '.join(preprocessed_fastq))
+
+    params = OrderedDict(sorted(parameters.items(), key=lambda t: t[0]))
+    params = ['--%s "%s"' % (k, v) if v is not True else '--%s' % k
+              for k, v in viewitems(params) if v]
+    cmd = 'deblur workflow --seqs-fp "%s" --output-dir "%s" %s' % (
+        preprocessed_fastq[0], out_dir, ' '.join(params))
```

Comment:
Yup! Here it is - note that the out_dir sent by Qiita should be used as working directory. In all the other plugins, the output dir for the command that you're calling should be creating in here. Thus, it should be something along the lines of `join(out_dir, "deblur_out")`. You can check pick otus and split libraries in other plugins.